deltacat/catalog/delegate.py

```diff
@@ -108,35 +102,33 @@ def create_table(
     namespace: Optional[str] = None,
     catalog: Optional[str] = None,
     lifecycle_state: Optional[LifecycleState] = None,
-    schema: Optional[Union[pa.Schema, str, bytes]] = None,
-    schema_consistency: Optional[Dict[str, SchemaConsistencyType]] = None,
-    partition_keys: Optional[List[Dict[str, Any]]] = None,
-    primary_keys: Optional[Set[str]] = None,
-    sort_keys: Optional[List[SortKey]] = None,
+    schema: Optional[Schema] = None,
+    partition_scheme: Optional[PartitionScheme] = None,
+    sort_keys: Optional[SortScheme] = None,
     description: Optional[str] = None,
-    properties: Optional[Dict[str, str]] = None,
-    permissions: Optional[Dict[str, Any]] = None,
+    table_properties: Optional[TableProperties] = None,
+    namespace_properties: Optional[NamespaceProperties] = None,
     content_types: Optional[List[ContentType]] = None,
-    replace_existing_table: bool = False,
+    fail_if_exists: bool = True,
     *args,
```

Comment:
Agreed. Only the table name is strictly required while every other optional can be defaulted, but it still leaves a question about how many of these are strictly necessary to expose to the end user at all. Namespace/catalog/table can also be rolled up into a fully qualified identifier string like `table="my_catalog.my_namespace.my_table"`, but IMO that just makes things worse (i.e., we're just saying the same things in a less explicit, more restricted way, and overall I'm in agreement with "Explicit is better than implicit" from [the Zen of Python](https://peps.python.org/pep-0020/)). 

Some of the apparent simplification here is a bit of an illusion too, in that removed arguments like `primary_keys` and `schema_consistency` just got nested within the new `Schema` type. Sometimes it makes sense to do that, but I wouldn't want to lean on that strategy too much since defining one or two very complex "TableCreateArgs" types may ultimately be much harder for the end user than just specifying 10 flat types that do the same thing (i.e., too many years of Java have also left me in firm agreement with "Flat is better than Nested" from the Zen of Python) 

Lifecycle state seems like a good early candidate to consider for removal though - it may be smart to always configure it to something like a "CREATED" state.

I think some of the answers here also require better understanding of the question, "When is a user creating an empty table that they're not immediately going to load with data anyways?"